tests/test_commands.py

```diff
@@ -236,9 +236,9 @@ def test_sql_comments(self):
         cmd._exec_and_print = MagicMock()
         cmd.process_iterable(sql.splitlines())
         cmd._exec_and_print.assert_has_calls([
-            call("SELECT 1"),
-            call("SELECT 2"),
-            call("SELECT /* this is a comment */ 3"),
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+            call("SELECT  3;"),
```

Comment:
I actually expected that the `sqlparse` library would filter out any SQL comments, what this adjustment reflects well.

At [^1], I mentioned:

> I think it may be essential that inline SQL comments get forwarded 1:1, because some SQL dialects accept annotations/decorations through specific inline SQL comment syntaxes, right?

So, it may be advisable to use `sqlparse`'s `strip_comments=False` option [^2] instead, if that would be possible?

P.S.: I think CrateDB does not use inline SQL comments for any purpose. I am merely looking at this detail from a general perspective [^3].

[^1]: https://github.com/crate/crash/pull/388#issuecomment-1595813170
[^2]: https://sqlparse.readthedocs.io/en/latest/api/#formatting-of-sql-statements
[^3]: Please advise or share your thoughts about this aspect, @seut, @matriv, and @hammerhead.

Reply:
> I think CrateDB does not use inline SQL comments for any purpose. I am merely looking at this detail from a general perspective.

Let me reference a fragment of a typical MySQL/MariaDB database dump file, created using the `mysqldump` command, which outlines corresponding inline SQL comments which are used to convey special commands/annotations/decorations.

<details>
<summary>mysqldump example</summary>

```sql
-- MySQL dump 10.16  Distrib 10.1.48-MariaDB, for debian-linux-gnu (i686)
--
-- Host: localhost    Database: test
-- ------------------------------------------------------
-- Server version	10.1.48-MariaDB-0+deb9u2

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;

--
-- Table structure for table `ccms_messages`
--

DROP TABLE IF EXISTS `ccms_messages`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `ccms_messages` (
  `id` int(11) NOT NULL,
  `sender` varchar(32) NOT NULL,
  `recievers` varchar(255) DEFAULT NULL,
  `msg_subject` varchar(255) DEFAULT NULL,
  `msg_body` text,
  `timestamp` int(11) DEFAULT NULL,
  `is_composed` tinyint(1) DEFAULT NULL,
  `is_new` tinyint(1) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=MyISAM DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `ccms_messages`
--

LOCK TABLES `ccms_messages` WRITE;
/*!40000 ALTER TABLE `ccms_messages` DISABLE KEYS */;
/*!40000 ALTER TABLE `ccms_messages` ENABLE KEYS */;
UNLOCK TABLES;
```

</details>

Reply:
Looking at [CrateDB's SQL Compatibility documentation](https://crate.io/docs/crate/reference/en/master/appendices/compatibility.html), I see you aim for PostgreSQL compatibility where such special comments do not exist as far as I know. 

If CrateDB does not use inline comments for any special purpose nor it tries to achieve MySQL/MariaDB full compatibility, I do not see a reason to keep sending the comments to the database.

I'm waiting for more input on this problem and I'll update the PR accordingly after.

Reply:
I agree with your assessment. Apologies for my suggestion, it was not appropriate.

> [Any kind of] comment is removed from the input stream before further syntax analysis and is effectively replaced by whitespace.
>
> -- https://www.postgresql.org/docs/8.0/sql-syntax.html#SQL-SYNTAX-COMMENTS

Reply:
Can you please adjust the comment [here](https://github.com/crate/crash/pull/386/files#diff-5592e672cef095f585fc0083c44cdfb8aafdfc11904a484dfa2b0794b0079f35R232) as inline comments are stripped out now?

Reply:
Thanks for spotting!

Reply:
Aha. At https://github.com/crate/crate/issues/13283, there is a proposal to indeed use SQL comments for sending hints to the execution engine. Maybe it is safer to use the `strip_comments=False` parser option right from the beginning?

Reply:
Well, I did not see that proposal. Taking it into account, `strip_comments=False` seems the right decision in order to avoid future inconveniences.

Reply:
Hi Mihail,

@matriv just confirmed your assessment at https://github.com/crate/crate/issues/13283#issuecomment-1600468021. So, may I ask you to adjust the patch to use `strip_comments=False`? Thank you so much!

With kind regards,
Andreas.

Reply:
So, can we please have some test(s) with inline, and multiline comments?
It's server-side job to treat them and shouldn't be stripped by crash.

Reply:
Hi Marios. Thank you for chiming in. I added corresponding test cases on behalf of GH-388 already, to set the stage, and asked @JustBeYou to adjust his patch accordingly. So, I think we agree on those matters, and it will have a good outcome.

Reply:
I know, just thought it would be nice to have also test with comments in the multi-line, multi-command context.

Reply:
As the conversation seems to have settled, I updated `strip_comments` to be false and added a few more test cases.
Notice that, `strip_comments=False` will keep all kinds of comments.